src/dvc_data/hashfile/meta.py

```diff
@@ -9,15 +10,50 @@ def _filter_default_or_none(field: Attribute, value: Any) -> bool:
 
 @define
 class Meta:
+    PARAM_ISDIR: ClassVar[str] = "isdir"
     PARAM_SIZE: ClassVar[str] = "size"
     PARAM_NFILES: ClassVar[str] = "nfiles"
     PARAM_ISEXEC: ClassVar[str] = "isexec"
     PARAM_VERSION_ID: ClassVar[str] = "version_id"
+    PARAM_ETAG: ClassVar[str] = "etag"
+    PARAM_CHECKSUM: ClassVar[str] = "checksum"
 
+    isdir: bool = False
     size: Optional[int] = None
     nfiles: Optional[int] = None
     isexec: bool = False
     version_id: Optional[str] = None
+    etag: Optional[str] = None
+    checksum: Optional[str] = None
+
+    @classmethod
+    def from_info(
+        cls, info: Dict[str, Any], protocol: Optional[str] = None
+    ) -> "Meta":
+        meta = Meta(
+            isdir=(info["type"] == "directory"),
+            size=info.get("size"),
+            isexec=is_exec(info.get("mode", 0)),
+            version_id=info.get("version_id"),
+        )
+
+        if protocol == "s3" and "ETag" in info:
+            meta.etag = info["ETag"].strip('"')
+        elif protocol == "gs" and "etag" in info:
+            import base64
+
+            meta.etag = base64.b64decode(info["etag"]).hex()
```

Comment:
This breaks azure support. adlfs returns `info['etag']` and did not need any modifications (so there were no if/else cases for azure).

But with the new changes we end up not setting either `meta.etag` or `meta.checksum`, since info does not contain `ETag` and fs is not s3 or gcs.

Also, do we even need to do the b64decode for gcs now? It seems like if we are going to treat etag/checksum as metadata, we should be saving it exactly as the fs gives it to us (without trying to convert it into something that looks kind of like an MD5).